core/modules/commands/listen.js

```diff
@@ -0,0 +1,48 @@
+/*\
+title: $:/core/modules/commands/listen.js
+type: application/javascript
+module-type: command
+
+Listen for HTTP requests and serve tiddlers
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+var Server = require("$:/core/modules/server/server.js").Server;
+
+exports.info = {
+	name: "listen",
+	synchronous: true,
+	namedParameterMode: true,
+	mandatoryParameters: [],
+};
+
+var Command = function(params,commander,callback) {
+	var self = this;
+	this.params = params;
+	this.commander = commander;
+	this.callback = callback;
+};
+
+Command.prototype.execute = function() {
+	var self = this;
+	if(!$tw.boot.wikiTiddlersPath) {
+		$tw.utils.warning("Warning: Wiki folder '" + $tw.boot.wikiPath + "' does not exist or is missing a tiddlywiki.info file");
+	}
+	// Set up server
+	this.server = new Server({
+		wiki: this.commander.wiki,
+		variables: self.params
+	});
+	var nodeServer = this.server.listen();
+	$tw.hooks.invokeHook("th-server-command-post-start",this.server,nodeServer);
```

Comment:
Should this be changed to `th-server-post-start` so it doesn't name a specific command? If so, the hook also needs to be changed in the "server" command. I think it is still soon enough to introduce this change. 

I'm still debating what to do with the `nodeServer` variable since I made this an event emitter on TiddlyServer, and it seems more logical for a plugin to create a new command rather than adding WebSockets to the existing "listen" command now that it is so ridiculously simple to duplicate. 

It would also work to send an event emitter instead of the NodeJS server and then emit a "server" event on the event emitter with the Node server object.  The plugin that adds websockets to the server would emit connect events on this same event emitter (`ws-client-connect`). This would make it easier for plugins to handle all scenarios without having to figure out which type of object they are getting (in the case of TiddlyServer, WebSocket connections are routed to the data folder instance they belong to using this event emitter).

Reply:
> Should this be changed to `th-server-post-start` so it doesn't name a specific command? 

I'd be inclined not to change it in the interests of backwards compatibility.

>  it seems more logical for a plugin to create a new command rather than adding WebSockets to the existing "listen" command now that it is so ridiculously simple to duplicate.

That approach can certainly be less brittle.

Reply:
> I'd be inclined not to change it in the interests of backwards compatibility.

I know. But this hasn't been around very long and I don't know if anyone is using it yet besides me. But its not a big deal.